hedera-node/hedera-app/src/main/java/com/hedera/node/app/throttle/HandleThrottleAccumulator.java

```diff
@@ -0,0 +1,491 @@
+        List<DeterministicThrottle> newActiveThrottles = new ArrayList<>();
+        EnumMap<HederaFunctionality, List<Pair<DeterministicThrottle, Integer>>> reqLists =
+                new EnumMap<>(HederaFunctionality.class);
+
+        for (var bucket : defs.throttleBuckets()) {
+            try {
+                final var utilThrottleBucket = new ThrottleBucket<>(
+                        bucket.burstPeriodMs(),
+                        bucket.name(),
+                        bucket.throttleGroups().stream()
+                                .map(this::hapiGroupFromPbj)
+                                .toList());
+                var mapping = utilThrottleBucket.asThrottleMapping(CAPACITY_SPLIT);
+                var throttle = mapping.getLeft();
+                var reqs = mapping.getRight();
+                for (var req : reqs) {
+                    reqLists.computeIfAbsent(req.getLeft(), ignore -> new ArrayList<>())
+                            .add(Pair.of(throttle, req.getRight()));
+                }
+                newActiveThrottles.add(throttle);
+            } catch (IllegalStateException badBucket) {
+                log.error("When constructing bucket '{}' from state: {}", bucket.name(), badBucket.getMessage());
+            }
+        }
+        EnumMap<HederaFunctionality, ThrottleReqsManager> newFunctionReqs = new EnumMap<>(HederaFunctionality.class);
+        reqLists.forEach((function, reqs) -> newFunctionReqs.put(function, new ThrottleReqsManager(reqs)));
+
+        functionReqs = newFunctionReqs;
+        activeThrottles = newActiveThrottles;
+
+        logResolvedDefinitions(CAPACITY_SPLIT);
+    }
+
+    public void applyGasConfig() {
+        long capacity;
+
+        final var configuration = configProvider.getConfiguration();
+        final var contractsConfig = configuration.getConfigData(ContractsConfig.class);
+        if (contractsConfig.throttleThrottleByGas() && contractsConfig.maxGasPerSec() == 0) {
+            log.warn("Consensus gas throttling enabled, but limited to 0 gas/sec");
+            return;
+        } else {
+            capacity = contractsConfig.maxGasPerSec();
+        }
+
+        gasThrottle = new GasLimitDeterministicThrottle(capacity);
+
+        final var configDesc = "Resolved consensus gas throttle -\n  "
+                + gasThrottle.capacity()
+                + " gas/sec (throttling "
+                + (contractsConfig.throttleThrottleByGas() ? "ON" : "OFF")
+                + ")";
+        log.info(configDesc);
+    }
+
+    private ThrottleGroup<HederaFunctionality> hapiGroupFromPbj(
+            com.hedera.hapi.node.transaction.ThrottleGroup pbjThrottleGroup) {
+        return new ThrottleGroup<>(pbjThrottleGroup.milliOpsPerSec(), pbjThrottleGroup.operations());
+    }
+
+    private void logResolvedDefinitions(int capacitySplit) {
+        var sb = new StringBuilder(
+                "Resolved handle throttles (after splitting capacity " + capacitySplit + " ways) - \n");
```

Comment:
It would be best not to concatenate a String when calling the constructor of `StringBuilder`. What is likely to happen is that the JVM will create a first StringBuilder to concatenate the String, and then the result is used to create a second StringBuilder. (At least, this happened the last time I analyzed such code a couple of years ago.)

We can check if the compiler became smarter and can identify this pattern. Or we can fix this manually:
```
var sb = new StringBuilder("Resolved handle throttles (after splitting capacity ").append(capacitySplit).append(" ways) - \n");
```

I know it is ugly, but the outcome is better. 🙂

Reply:
Fixed.